packages/repository-json-schema/src/build-schema.ts

```diff
@@ -64,40 +83,55 @@ export function stringTypeToWrapper(type: string): Function {
  * @param ctor Constructor
  */
 export function isComplexType(ctor: Function) {
-  return !([String, Number, Boolean, Object, Function] as Function[]).includes(
-    ctor,
-  );
+  return !([
+    String,
+    Number,
+    Boolean,
+    Object,
+    Function,
+    Array,
+  ] as Function[]).includes(ctor);
+}
+
+/**
+ * Determines whether a given string or constructor is array type or not
+ * @param type Type as string or wrapper
+ */
+export function isArrayType(type: string | Function) {
+  return type === Array || type === 'array';
 }
 
 /**
  * Converts property metadata into a JSON property definition
  * @param meta
  */
 export function metaToJsonProperty(meta: PropertyDefinition): JSONSchema {
-  let ctor = meta.type as string | Function;
-  let def: JSONSchema = {};
-
-  // errors out if @property.array() is not used on a property of array
-  if (ctor === Array) {
-    throw new Error('type is defined as an array');
-  }
-
-  if (typeof ctor === 'string') {
-    ctor = stringTypeToWrapper(ctor);
+  // tslint:disable-next-line:no-any
+  const propDef: JSONSchema = {};
+  let result: JSONSchema;
+  let propertyType = meta.type as string | Function;
+
+  if (isArrayType(propertyType) && meta.itemType) {
```

Comment:
hmm...any reason why using two different ways to assert the array type?
`isArrayType` vs `Array.isArray`

Reply:
`isArrayType` essentially asserts whether given data corresponds to the Wrapper `Array` or the string 'array'